src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
     (ActorFunc)ObjWarp2block_Update,
     (ActorFunc)ObjWarp2block_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Warp2block/func_80BA1DF0.s")
+typedef struct {
+  f32 scale;
+  f32 focus;
+  s16 params;
+} Warp2BlockSpawnData;
+
+ Warp2BlockSpawnData sSpawnData[2] = {
+	{ 1.0f, 60.0f, 0x0018 },
+	{ 0.6f, 40.0f, 0x0019 },
+};
+
+f32 sDistances[] = {60.0f, 100.0f, 140.0f, 180.0f, 220.0f, 260.0f, 300.0f, 300.0f};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1800, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 300, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1500, ICHAIN_STOP),
+};
+
+Color_RGB8 sColors[] = {
+	{0x64, 0x78, 0x8C}, {0x50, 0x8C, 0xC8}, {0x64, 0x96, 0xC8}, {0x64, 0xC8, 0xF0},
+	{0x50, 0x6E, 0x8C}, {0x46, 0xA0, 0xE1}, {0x50, 0x64, 0x82}, {0x64, 0x6E, 0xBE},
+};
+
+void ObjWarp2block_Spawn(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, (u16)0x8B, 
+    	this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 
+    	0, 0, 0, 
+    	sSpawnData[(this->dyna.actor.params >> 8) & 1].params);
+
+    Actor_Spawn(&globalCtx->actorCtx, globalCtx, (u16)0x8B, 
```

Comment:
These actor IDs need to be macros.

Reply:
You mean the 0x8B part? From other uses of Actor_Spawn it looks like they're values in ActorID (https://github.com/zeldaret/oot/blob/00a5edea71a605e7bdad78892b2863000e53624e/include/z64actor.h#L779), which would make this usage here ACTOR_DEMO_EFFECT.

Is that correct?

Reply:
yeah, that's the one.

Reply:
Sweet, thanks! Done.